src/sizes.test.tsx

```diff
@@ -0,0 +1,70 @@
+/** @jsx jsx */
+// ----- Imports ----- //
+
+import renderer from 'react-test-renderer';
+import { breakpoints } from '@guardian/src-foundations/mq';
+import { jsx } from '@emotion/core';
+import { matchers } from 'jest-emotion';
+
+import { sizesAttribute, Sizes, styles } from './sizes';
+
+
+// ----- Setup ----- //
+
+expect.extend(matchers);
+
+const noQueries = { mediaQueries: [], default: '30vw' };
+const sizes: Sizes = {
+    mediaQueries: [
+        { breakpoint: 'desktop', size: '20px' },
+        { breakpoint: 'tablet', size: '20%' },
```

Comment:
I'm very much of the opinion that percentage values should be avoided, because, as mentioned, they relate to the containing block **height**.

Reply:
> percentage values should be avoided
> For a subsequent PR?

Does that mean that we ask consumers to avoid them, or that we allow consumers to use them but figure out some way to handle that? The downside I can see to the former is that perhaps it's valid to use a percentage for `width` - it's a consequence of our implementation that this will also affect the height, not the consumer's fault.

Either way, if we decide to prevent the consumer using them we could implement that more sophisticated type for size lengths. That way we can restrict percentages at compile-time.